### DISCUSSION TITLE
Support for other databases

### DISCUSSION THREAD

**Discussion by riazarbi** [brooklyn-data/dbt_artifacts#347] [Q&A]
Hello, 

I want to implement something similar to this package for duckdb. Is there any particular reason only a small selection of databases are supported? What's stopping me from just running this against, for instance, duckdb?

**Comment by jared-rimmer** [comment]
Hi @riazarbi 👋 thanks for opening this discussion.

Supporting a new database is just a case of ensuring that the correct syntax and methods are used for the database in question.

You will see through looking at the code base (mainly `macros`) that we have different macros for BigQuery vs Snowflake / Databricks.

Supporting duckdb would be a case of determining whether the existing macros work and / or making the necessary changes to macros so that it does.

If you're willing to look into this we would always be willing to accept a PR
